app/assets/javascripts/sidebar/sidebar.controller.js

```diff
@@ -5,30 +5,45 @@
 angular.module('GeoBox')
 .controller('SidebarController', SidebarController);
 
-SidebarController.$inject = ['$rootScope', 'DocumentService', '$scope'];
-function SidebarController($rootScope, DocumentService, $scope){
+SidebarController.$inject = ['$rootScope', 'DocumentService', '$scope', 'UserService'];
+function SidebarController($rootScope, DocumentService, $scope, UserService){
 	var $ctrl = this;
 	$ctrl.docs = {};
-
+	$ctrl.user = {};
+	$ctrl.base_url = null;
 
 	$ctrl.$onInit = function(){
 		$rootScope.$broadcast('login:login', {}); // we can consider adding data to this event
 		var sidebar = $('#sidebar').sidebar();
+		UserService.getUserInfo();
 	};
 
 	$ctrl.getDocs = function(e,d){
 		$ctrl.docs = processDocuments(DocumentService.docs);
 	};
 
-	$scope.$on('documents:ready', $ctrl.getDocs);
+	$ctrl.getUser = function(e,d){
```

Comment:
where is getuser called? we used to have scope onready but now we just have ctrl.getUser? i am not seeing the dataflow

Reply:
$ctrl.getUser is a mirror image of $ctrl.getDocs, it's a callback for when the 'user:ready' event is heard on the controller scope

ah, i moved the scope lines to the bottom. for some reason, the callbacks given to scope needed to be defined before they are used. i'm not sure why this wasn't causing problems earlier with getDocs.

Reply:
ahh i see